clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/ops/DeployCloudFoundryServerGroupAtomicOperation.java

```diff
@@ -472,28 +473,75 @@ private void scaleApplication(
         convertToMb("memory", description.getApplicationAttributes().getMemory());
     Integer diskSizeAmount =
         convertToMb("disk quota", description.getApplicationAttributes().getDiskQuota());
-
     client
         .getProcesses()
         .scaleProcess(
             serverGroupId,
             description.getApplicationAttributes().getInstances(),
             memoryAmount,
             diskSizeAmount);
+
+    if (!description.getApplicationAttributes().getProcesses().isEmpty()) {
+      List<Process> processes = client.getProcesses().getAllProcessesByAppId(serverGroupId);
+
+      for (ProcessRequest req : description.getApplicationAttributes().getProcesses()) {
+        String processGuid =
+            processes.stream()
+                .filter(p -> p.getType().equalsIgnoreCase(req.getType()))
+                .map(p -> p.getGuid())
+                .findFirst()
+                .orElseThrow(
+                    () ->
+                        new CloudFoundryApiException(
+                            "Unable to find a process with type: " + req.getType()));
+
+        Integer pMemoryAmount = convertToMb("memory", req.getMemory());
+        Integer pDiskSizeAmount = convertToMb("disk quota", req.getDiskQuota());
+        client
+            .getProcesses()
+            .scaleProcess(processGuid, req.getInstances(), pMemoryAmount, pDiskSizeAmount);
+      }
+    }
     getTask().updateStatus(PHASE, "Scaled application '" + description.getServerGroupName() + "'");
   }
 
   private void updateProcess(
       String serverGroupId, DeployCloudFoundryServerGroupDescription description) {
     CloudFoundryClient client = description.getClient();
     getTask().updateStatus(PHASE, "Updating process '" + description.getServerGroupName() + "'");
+
     client
         .getProcesses()
         .updateProcess(
             serverGroupId,
             description.getApplicationAttributes().getCommand(),
             description.getApplicationAttributes().getHealthCheckType(),
             description.getApplicationAttributes().getHealthCheckHttpEndpoint());
+
+    if (!description.getApplicationAttributes().getProcesses().isEmpty()) {
+      List<Process> processes = client.getProcesses().getAllProcessesByAppId(serverGroupId);
+
+      for (ProcessRequest req : description.getApplicationAttributes().getProcesses()) {
+        String processGuid =
+            processes.stream()
```

Comment:
This stream looks very similar to the one from line 489. Is it worth extracting it to a method?

Reply:
Its pretty small and not doing anything logic-wise but I like where your head is at 👍 . Done!